openmc/mesh.py

```diff
@@ -586,3 +588,129 @@ def to_xml_element(self):
         subelement.text = ' '.join(map(str, self.z_grid))
 
         return element
+
+
+class UnstructuredMesh(MeshBase):
+    """A 3D unstructured mesh
+
+    Parameters
+    ----------
+    mesh_id : int
+        Unique identifier for the mesh
+    name : str
+        Name of the mesh
+    filename : str
+        Location of the unstructured mesh file
+
+    Attributes
+    ----------
+    id : int
+        Unique identifier for the mesh
+    name : str
+        Name of the mesh
+    mesh_file : str
```

Comment:
```suggestion
    filename : str
```
Also, it does seem slightly odd to me that in the HDF5 file we are using `mesh_file` whereas here we use `filename`.

Reply:
True. I'll update that to be more consistent.